### PR TITLE
Fix `FunctionNode.get_retained_outputs` to return `()` if no output is retained

### LINKED ISSUE #6117: FunctionNode.get_retained_outputs returns None if retain_outputs is not called
`FunctionNode.get_retained_outputs` returns `None` if `retain_outputs` is not called.

```py
import numpy
import chainer


class MyFunc(chainer.FunctionNode):
    def forward(self, inputs):
        x1, = inputs
        #self.retain_outputs(())
        return x1 * 3,

    def backward(self, input_indices, grad_outputs):
        outputs = self.get_retained_outputs()
        print(outputs)
        gy, = grad_outputs
        return gy * 3,


x1 = chainer.Variable(numpy.array([1, 2], numpy.float32))
y, = MyFunc().apply((x1,))

y.grad = numpy.ones_like(y.array)
y.backward()
```
```
None
```

If `retain_outputs` is called, it becomes `()`.
OTOH `retain_inputs` is consistent.

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #6117

### CODE REVIEW

**Comment by hvy** [comment]
Jenkins failures seem unrelated,

File: tests/chainer_tests/test_function_node.py [hunk 1]
```diff
@@ -744,16 +756,39 @@ def check_function_node_retain(self, xp):
...
+                x, = inputs
+                if not skip_call:
+                    self.retain_outputs(())
+                    self.retain_inputs(())
+                return x * 3,
+
+            def backward(self, input_indices, grad_outputs):
+                self.backward_called += 1
+                assert self.get_retained_outputs() is ()
+                assert self.get_retained_inputs() is ()
```
[the remark this answers is not in the document]
Comment (niboshi) [reply tests/chainer_tests/test_function_node.py]: Oh, I didn't know that

[the remark this answers is not in the document]
Comment (niboshi) [reply tests/chainer_tests/test_function_node.py]: Fixed.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
Shouldn't we backport?

**Comment by hvy** [comment]
I tried to backport it but noticed it's rather not straight forward. For instance, this is not backported https://github.com/chainer/chainer/commit/7c7f97374dc59dfd94dda9ad4944b8cd61a3eb30#diff-3480da470ca3b18f6b879875999e6206L757

**Comment by niboshi** [comment]
@hvy 
You can redo the fix to the input retention manually.

**Pull request merged**